app/services/performance.ts

```diff
@@ -1,43 +1,99 @@
 import Vue from 'vue';
-import { Subject } from 'rxjs';
-
-import { StatefulService, mutation } from './core/stateful-service';
+import { Subject, Subscription } from 'rxjs';
 import electron from 'electron';
 import * as obs from '../../obs-api';
+import { StatefulService, mutation, Service, Inject } from 'services';
+import { throttle } from 'lodash-decorators';
+import {
+  NotificationsService,
+  ENotificationType,
+  ENotificationSubType,
+} from 'services/notifications';
+import { ServicesManager } from '../services-manager';
+import { JsonrpcService } from './api/jsonrpc';
+import { TroubleshooterService, TIssueCode } from 'services/troubleshooter';
+import { $t } from 'services/i18n';
 
 interface IPerformanceState {
   CPU: number;
   numberDroppedFrames: number;
   percentageDroppedFrames: number;
+  numberSkippedFrames: number;
+  percentageSkippedFrames: number;
+  numberLaggedFrames: number;
+  percentageLaggedFrames: number;
+  numberEncodedFrames: number;
+  numberRenderedFrames: number;
   bandwidth: number;
   frameRate: number;
 }
 
-const STATS_UPDATE_INTERVAL = 2 * 1000;
+interface INextStats {
+  framesSkipped: number;
+  framesEncoded: number;
+  skippedFactor: number;
+  framesLagged: number;
+  framesRendered: number;
+  laggedFactor: number;
+  droppedFramesFactor: number;
+}
+
+export enum EStreamQuality {
+  GOOD = 'GOOD',
+  FAIR = 'FAIR',
+  POOR = 'POOR',
+}
+
+// How frequently parformance stats should be updated
+const STATS_UPDATE_INTERVAL = 5 * 1000;
+// Limit on interval between unique notification types
+const NOTIFICATION_THROTTLE_INTERVAL = 2 * 60 * 1000;
+// Time window for averaging notification issues
+const SAMPLING_DURATION = 2 * 60 * 1000;
 
-// TODO: merge this service with PerformanceMonitorService
+interface IMonitorState {
+  framesLagged: number;
+  framesRendered: number;
+  framesSkipped: number;
+  framesEncoded: number;
+}
 
 // Keeps a store of up-to-date performance metrics
 export class PerformanceService extends StatefulService<IPerformanceState> {
+  @Inject() private notificationsService: NotificationsService;
+  @Inject() private jsonrpcService: JsonrpcService;
+  @Inject() private troubleshooterService: TroubleshooterService;
+
   static initialState: IPerformanceState = {
     CPU: 0,
     numberDroppedFrames: 0,
     percentageDroppedFrames: 0,
+    numberSkippedFrames: 0,
+    percentageSkippedFrames: 0,
+    numberLaggedFrames: 0,
+    percentageLaggedFrames: 0,
```

Comment:
do we need percentages stored in state? i feel like those could all be calculated values

Reply:
Can refactor this I think

Reply:
So percentage dropped frames comes directly from OBS pre-calculated.
I can make the other two percentages getters with calculated values but I think it's better to keep things consistent, unless there's a good reason to avoid storing things in state as much as possible.